### ISSUE TITLE
How to use Nprinting-Cli or Nprinting API to perform 'Add Engine'?

### ISSUE THREAD

**Issue by Kev-kutkin** [Nillth/Qlik-NPrinting-CLI#2]
I would like to add new Engines to the NPrinting under administrator credentials. However, I could not find any relevant information from the NPrinting API reference site and the Module from NPrinting-Cli. Can I get guidance  using API or the NPrinting-Cli to

1. to create new Engine;
2. to update the existing Engine's configurations

I am not sure how to get the ID for the engine when using 
`Invoke-NPRequest -NPE engines -Method POST ......`
Another thing is that tt seems I need to perform authentication' and filter' s methods prior to Posting. 

Thank you for your advice.

Regards,
Kevin

**Comment by Nillth** [comment]
you cannot perform any action without first authenticating against the API's using Connect-NPrinting first

**Comment by Nillth** [comment]
When you install the NPrinting Engine, there is a step where you have to specify the engine Password

![NPrintingEngineSetup](https://user-images.githubusercontent.com/22066262/148015464-b9c5b8c3-ff83-4169-8ac5-e20a30ebfd01.png)

Update the Variable `$ENGINE_INSTALL_PASSWORD` with the value used during the install process
If the NPrinting Engine is not installed on the same server as the NPrinting Webconsole the varaiables `$ENGINE_NAME` & `$ENGINE_HOST` may also need to be updated.

It is assumed that the account you are logged in with that is running the PowerShell session will be able to authenticate using windows credentials to the NPrinting API's if not you will need to adapt the `Connect-NPrinting` command to suit your environment.

``` PowerShell
$ENGINE_INSTALL_PASSWORD = "qlik"
$ENGINE_NAME = "$($env:COMPUTERNAME)" 
$ENGINE_HOST = "$($env:COMPUTERNAME)"

$Engine = @{
    enabled = $true
    name = $ENGINE_NAME
    host = $ENGINE_HOST
}

Connect-NPrinting -TrustAllCerts

$NewEngine = Invoke-NPRequest -path "engines" -method Post -Data $Engine -NPE
$EngineInstallPass = @{ "password" = $ENGINE_INSTALL_PASSWORD }
$NewEngine = Invoke-NPRequest -path "engines/$($NewEngine.id)/certificate" -NPE -method Post -Data $EngineInstallPass
```

**Comment by Kev-kutkin** [comment]
Hi @Nillth,

Thank you for your great guidance. I have followed your steps, but I encountered error in the step below.
`$NewEngine = Invoke-NPRequest -path "engines" -method Post -Data $Engine -NPE`

The error indicates that the Remote server returned an error: (500) Internal Server Error.  
I therefore share my -verbose result for the above code in the picture below and hope to get further guidance. 

![error_engine_script](https://user-images.githubusercontent.com/25189239/148393561-26d600a5-8788-4c54-a5fa-a91712e78e8b.jpeg)


Thank you for your guidance, @Nillth
